src/components/moderation/purge.ts

```diff
@@ -2,27 +2,37 @@ import * as Discord from "discord.js";
 
 import { strict as assert } from "assert";
 
-import { M } from "../utils/debugging-and-logging.js";
-import { DAY, HOUR, MINUTE, colors } from "../common.js";
-import { BotComponent } from "../bot-component.js";
-import { CommandSetBuilder } from "../command-abstractions/command-set-builder.js";
-import { Wheatley } from "../wheatley.js";
-import { EarlyReplyMode, TextBasedCommandBuilder } from "../command-abstractions/text-based-command-builder.js";
-import { CommandAbstractionReplyOptions, TextBasedCommand } from "../command-abstractions/text-based-command.js";
-import { duration_regex, parse_duration } from "./moderation/moderation-common.js";
-import { pluralize } from "../utils/strings.js";
-import { SelfClearingMap } from "../utils/containers.js";
-import { url_re } from "./quote.js";
-import { ascending, unwrap } from "../utils/misc.js";
-import { MessageContextMenuInteractionBuilder } from "../command-abstractions/context-menu.js";
-import { decode_snowflake, discord_timestamp, forge_snowflake } from "../utils/discord.js";
-import { message_database_entry } from "../infra/schemata/logged-messages.js";
-import { chunks } from "../utils/arrays.js";
+import { M } from "../../utils/debugging-and-logging.js";
+import { DAY, HOUR, MINUTE, colors } from "../../common.js";
+import { BotComponent } from "../../bot-component.js";
+import { CommandSetBuilder } from "../../command-abstractions/command-set-builder.js";
+import { Wheatley } from "../../wheatley.js";
+import { EarlyReplyMode, TextBasedCommandBuilder } from "../../command-abstractions/text-based-command-builder.js";
+import { CommandAbstractionReplyOptions, TextBasedCommand } from "../../command-abstractions/text-based-command.js";
+import { duration_regex, parse_duration } from "../moderation/moderation-common.js";
+import { pluralize } from "../../utils/strings.js";
+import { SelfClearingMap } from "../../utils/containers.js";
+import { url_re } from "../quote.js";
+import { ascending, unwrap } from "../../utils/misc.js";
+import { MessageContextMenuInteractionBuilder } from "../../command-abstractions/context-menu.js";
+import { decode_snowflake, discord_timestamp, forge_snowflake } from "../../utils/discord.js";
+import { chunks } from "../../utils/arrays.js";
 
 type PurgableChannel = Exclude<Discord.TextBasedChannel, Discord.DMChannel | Discord.PartialDMChannel>;
 type PurgableMessages = Discord.Collection<string, Discord.Message> | string[];
 type PurgeWork = [PurgableChannel, Iterable<PurgableMessages> | AsyncGenerator<PurgableMessages>];
 
+export type message_database_entry = {
```

Comment:
I am concerned with this falling out of line with the more complete definition, unless that's what you were asking about asserting the other day

Reply:
I assume you saw what I did in the other place to take care of that?